crates/noirc_evaluator/src/ssa/acir_gen/acir_mem.rs

```diff
@@ -123,6 +118,81 @@ impl ArrayHeap {
         self.staged.insert(index, (value, op));
     }
 
+    /// This helper function transforms an expression into a linear expression, by generating a witness if the input expression is not linear nor constant
```

Comment:
This isn't complete as we also require that expression is univariate rather than just linear.

Reply:
Yes it is correct, if the linear term has more than one element, it is not a linear expression but a multilinear expression

Reply:
Huh, I've never seen a requirement to for an expression to only have a single variable to be a linear expression. I think calling out the special case as univariate is clearer than having the term "multilinear". Not a huge issue however.

Reply:
Yet, this is the proper terminology. I have nonetheless added the univariate term to make it clearer.